tests/SpanProcessorTest.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\RST;
+
+use Doctrine\RST\Environment;
+use Doctrine\RST\Span\SpanProcessor;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class SpanProcessorTest extends TestCase
+{
+    /** @var Environment|MockObject */
+    private $environment;
+
+    protected function setUp(): void
+    {
+        $this->environment = $this->createMock(Environment::class);
+    }
+
+    public function testProcessingLinkSpanSetsLinkTarget(): void
+    {
+        $this->environment->expects(self::exactly(1))->method('setLinkTarget');
+        $spanProcessor = new SpanProcessor($this->environment, '`example <http://example.org/>`_');
```

Comment:
I'd feel more secure about that test when `setLinkTarget` confirms that it gets the expected value.

Reply:
setLinkTarget gets called with an object of LinkTarget. I didn't find a way to test that and thought it is covered in the Functional tests then. Any idea?

Reply:
@SenseException , maybe like this with a callback?

Reply:
I think a simple instance creation of the object with the same expected constructor values should do the trick too. AFAIK objects are compared with `==` in `with()`
https://3v4l.org/GBLYU

Reply:
Cool, didn't know that